common/src/main/java/com/wynntils/features/overlays/GameBarsOverlayFeature.java

```diff
@@ -0,0 +1,580 @@
+                    81,
+                    healthTexture.get().getTextureY2(),
+                    progress);
+        }
+    }
+
+    public static class BloodPoolBarOverlay extends HealthBarOverlay {
+        protected BloodPoolBarOverlay() {
+            super(
+                    new OverlayPosition(
+                            -30,
+                            -150,
+                            VerticalAlignment.Bottom,
+                            HorizontalAlignment.Center,
+                            OverlayPosition.AnchorSection.BottomMiddle),
+                    new OverlaySize(81, 21));
+        }
+
+        @Override
+        public String icon() {
+            return "⚕";
+        }
+
+        @Override
+        public BossBarProgress progress() {
+            return Models.BossBar.bloodPoolBar.getBarProgress();
+        }
+
+        @Override
+        public boolean isActive() {
+            return Models.BossBar.bloodPoolBar.isActive();
+        }
+
+        @Override
+        protected void onConfigUpdate(ConfigHolder configHolder) {
+            // Do not call super
+        }
+    }
+
+    protected static class ManaBarOverlay extends BaseBarOverlay {
+        @RegisterConfig("overlay.wynntils.manaBar.manaTexture")
+        public final Config<ManaTexture> manaTexture = new Config<>(ManaTexture.a);
+
+        protected ManaBarOverlay() {
+            this(
+                    new OverlayPosition(
+                            -30,
+                            52,
+                            VerticalAlignment.Bottom,
+                            HorizontalAlignment.Center,
+                            OverlayPosition.AnchorSection.BottomMiddle),
+                    new OverlaySize(81, 21));
+        }
+
+        protected ManaBarOverlay(OverlayPosition overlayPosition, OverlaySize OverlaySize) {
+            super(overlayPosition, OverlaySize, CommonColors.LIGHT_BLUE);
+        }
+
+        @Override
+        public float textureHeight() {
+            return manaTexture.get().getHeight();
+        }
+
+        @Override
+        public BossBarProgress progress() {
+            CappedValue mana = Models.CharacterStats.getMana();
+            return new BossBarProgress(mana, (float) mana.getProgress());
+        }
+
+        @Override
+        public String icon() {
+            return "✺";
+        }
+
+        @Override
+        public boolean isActive() {
+            return true;
+        }
+
+        @Override
+        protected void onConfigUpdate(ConfigHolder configHolder) {
+            Models.CharacterStats.hideMana(this.shouldBeEnabled() && !this.shouldDisplayOriginal.get());
+        }
+
+        @Override
+        protected void renderBar(
+                PoseStack poseStack,
+                MultiBufferSource.BufferSource bufferSource,
+                float renderY,
+                float renderHeight,
+                float progress) {
+            if (progress > 1) { // overflowing mana
```

Comment:
Why do we need to implement renderBar here? We're doing something the superclass can't handle?

Reply:
It's overflowing code, I did not invent these, so I did not touch it.

Reply:
(remember that i just copied these)

Reply:
Ok. One more thing we can have a look at for future improvements, see if this code can be moved upwards.